server/controllers/contest.js

```diff
@@ -60,3 +62,31 @@ exports.getContests = asyncHandler(async (req, res, next) => {
     throw new Error('failed to get contests');
   }
 });
+
+// handler for create submission by contest id
+exports.createSubmissionByContestId = asyncHandler(async (req, res, next) => {
+  const contestId = req.params.id;
+  const s3UrlArray = req.body.data;
+  const userId = req.user.id;
+  if (!ObjectId.isValid(userId)) {
+    return res.status(400).json({
+      error: 'User ID is invalid.',
+    });
+  }
+  try {
+    const submissionData = await Submission.create({
```

Comment:
Looks good! Maybe you would also want to check to see if a submission from the current user already exists? If yes, update the existing submission. If no, then maybe create a new one? :)

Reply:
add some function to check userId and contestId already existed in submission._id, if Yes, update the files in database, otherwise, create a new one.